### PR TITLE
feat(aio): set the document title in the browser title

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

**Review by Foxandxss — CHANGES_REQUESTED** [review]
Other things than docs gives undefined.

**Comment by petebacondarwin** [comment]
Good point. I'll fix it up

**Comment by petebacondarwin** [comment]
By ensuring `doc.title` is never undefined

**Review by wardbell — APPROVED** [review]
LGTM

File: aio/tools/transforms/angular.io-package/processors/convertToJson.js [hunk 1]
```diff
@@ -8,13 +8,27 @@ module.exports = function convertToJsonProcessor() {
...
-            title: doc.title || doc.name,
-            contents: doc.renderedContent
-          };
-          doc.renderedContent = JSON.stringify(output, null, 2);
+          let title = doc.title || doc.name;
+          let contents = doc.renderedContent || '';
+
+          // If there is no title then try to extract it from the first h1 in the renderedContent
+          if (title === undefined) {
+            contents = contents.replace(/<h1[^>]*>(.+?)<\/h1>/, (outerHeading, innerHeading) => {
```
Comment (gkalpak) [inline aio/tools/transforms/angular.io-package/processors/convertToJson.js]: Why `replace` when you are not actually replacing it? I assume it is a left-over from a previous implementation that _did_ replace it.

Comment (petebacondarwin) [inline aio/tools/transforms/angular.io-package/processors/convertToJson.js]: you're right. I'll change that

**Comment by petebacondarwin** [comment]
@gkalpak PTAL

File: aio/tools/transforms/angular.io-package/processors/convertToJson.spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  it('should set missing titles to empty', () => {
+    const docs = [ { docType: 'test-doc' }];
+    processor.$process(docs);
+    expect(JSON.parse(docs[0].renderedContent).title).toBe('');
+  });
+
+  it('should log a warning', () => {
+    const docs = [ { docType: 'test-doc' }];
```
Comment (gkalpak) [inline aio/tools/transforms/angular.io-package/processors/convertToJson.spec.js]: Consistently inconsistent spacing (`[ {` vs `}]`).
<sub>(Which is way better than inconsistently inconsistent spacing, but still... 😛 </sub>

**Review by gkalpak — APPROVED** [review]
LGTM

File: aio/tools/transforms/angular.io-package/processors/convertToJson.spec.js (hunk 2 above)
Comment (petebacondarwin) [inline aio/tools/transforms/angular.io-package/processors/convertToJson.spec.js]: :-(

**Pull request merged**